streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -229,37 +230,35 @@ public Subscription subscription(Set<String> topics) {
         if (internalTopicManager != null) {
             log.debug("Starting to validate internal source topics in partition assignor.");
 
-            if (internalSourceTopicToTaskIds != null) {
-                for (Map.Entry<String, Set<TaskId>> entry : internalSourceTopicToTaskIds.entrySet()) {
-                    String topic = streamThread.jobId + "-" + entry.getKey();
+            for (Map.Entry<String, Set<TaskId>> entry : internalSourceTopicToTaskIds.entrySet()) {
+                String topic = streamThread.jobId + "-" + entry.getKey();
 
-                    // should have size 1 only
-                    int numPartitions = -1;
-                    for (TaskId task : entry.getValue()) {
-                        numPartitions = task.partition;
-                    }
+                // should have size 1 only
+                int numPartitions = -1;
+                for (TaskId task : entry.getValue()) {
+                    numPartitions = task.partition;
+                }
 
-                    internalTopicManager.makeReady(topic, numPartitions);
+                internalTopicManager.makeReady(topic, numPartitions);
 
-                    // wait until the topic metadata has been propagated to all brokers
-                    List<PartitionInfo> partitions;
-                    do {
-                        partitions = streamThread.restoreConsumer.partitionsFor(topic);
-                    } while (partitions == null || partitions.size() != numPartitions);
+                // wait until the topic metadata has been propagated to all brokers
+                List<PartitionInfo> partitions;
+                do {
+                    partitions = streamThread.restoreConsumer.partitionsFor(topic);
+                } while (partitions == null || partitions.size() != numPartitions);
 
-                    metadata.update(topic, partitions);
-                }
+                metadata.update(topic, partitions);
             }
 
             log.info("Completed validating internal source topics in partition assignor.");
         }
+        internalSourceTopicToTaskIds.clear();
```

Comment:
Hmm, a few lines further down in the same `assign()` method we have:

```java
            for (String topicName : topicGroups.get(task.topicGroupId).interSourceTopics) {
                Set<TaskId> tasks = internalSourceTopicToTaskIds.get(topicName);
                if (tasks == null) {
                    tasks = new HashSet<>();
                    internalSourceTopicToTaskIds.put(topicName, tasks);
                }

                tasks.add(task);
            }
```
If we `clear()` the hash map here (highlighted line on GH), then the code snippet above will always return null.